src/tmt_web/api.py

```diff
@@ -0,0 +1,163 @@
+                description="Reference of the test repository",
+            ),
+        ] = None,
+        test_path: Annotated[
+            str | None,
+            Query(
+                alias="test-path",
+                title="Test path",
+                description="Path to the test metadata directory",
+            ),
+        ] = None,
+        plan_url: Annotated[
+            str | None,
+            Query(
+                alias="plan-url",
+                title="Plan URL",
+                description="URL of a Git repository containing plan metadata",
+            ),
+        ] = None,
+        plan_name: Annotated[
+            str | None,
+            Query(
+                alias="plan-name",
+                title="Plan name",
+                description="Name of the plan",
+            ),
+        ] = None,
+        plan_ref: Annotated[
+            str | None,
+            Query(
+                alias="plan-ref",
+                title="Plan reference",
+                description="Reference of the plan repository",
+            ),
+        ] = None,
+        plan_path: Annotated[
+            str | None,
+            Query(
+                alias="plan-path",
+                title="Plan path",
+                description="Path to the plan metadata directory",
+            ),
+        ] = None,
+        out_format: Annotated[Literal["html", "json", "yaml"], Query(alias="format")] = "json",
+) -> TaskOut | str | Any:
+    # Parameter validations
+    if (test_url is None and test_name is not None) or (test_url is not None and test_name is None):
+        return "Invalid arguments!"
+    if (plan_url is None and plan_name is not None) or (plan_url is not None and plan_name is None):
+        return "Invalid arguments!"
+    if plan_url is None and plan_name is None and test_url is None and test_name is None:
+        return "Missing arguments!"
+        # TODO: forward to docs
+    # Disable Celery if not needed
+    if os.environ.get("USE_CELERY") == "false":
+        return service.main(
+            test_url=test_url,
+            test_name=test_name,
+            test_ref=test_ref,
+            plan_url=plan_url,
+            plan_name=plan_name,
+            plan_ref=plan_ref,
+            out_format=out_format,
+            test_path=test_path,
+            plan_path=plan_path)
+    r = service.main.delay(
+            test_url=test_url,
+            test_name=test_name,
+            test_ref=test_ref,
+            plan_url=plan_url,
+            plan_name=plan_name,
+            plan_ref=plan_ref,
+            out_format=out_format,
+            test_path=test_path,
+            plan_path=plan_path)
+    # Special handling of response if the format is html
+    if out_format == "html":
+        status_callback_url = f'{os.getenv("API_HOSTNAME")}/status/html?task-id={r.task_id}'
+        return HTMLResponse(content=html_generator.generate_status_callback(r, status_callback_url))
+    else:
```

Comment:
I'm pointing it out constantly, and it would be nice to have a linter to do so for me. I even though we have it enabled, I recall `ruff` refactoring a couple of conditions here and there.

Reply:
[`RET505`](https://docs.astral.sh/ruff/rules/superfluous-else-return/). It's in the `lint.ignore` list, probably copied form `tmt` :sweat_smile:

Reply:
> I'm pointing it out constantly, and it would be nice to have a linter to do so for me. I even though we have it enabled, I recall `ruff` refactoring a couple of conditions here and there.

@happz so if I understand it correctly you'd like to enforce this linter check? With this being more-or-less clean slate, we can do things like this, I assume.  
 Incl. single vs. double quotes :P